src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -835,34 +851,30 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ, colidx)
                 )
             ∂ᶠu₃ʲ_err_∂ᶠu₃ʲ =
                 matrix[@name(f.sgsʲs.:(1).u₃), @name(f.sgsʲs.:(1).u₃)]
-            ᶜu₃ʲ = p.ᶜtemp_C3
+            ᶜu₃ʲ = ᶜtemp_C3
             @. ᶜu₃ʲ[colidx] = ᶜinterp(Y.f.sgsʲs.:(1).u₃[colidx])
+            tmp = ᶠtridiagonal_matrix_c3
+            @. tmp[colidx] =
+                -(ᶠgradᵥ_matrix()) ⋅ convert(
+                    BidiagonalMatrixRow{FT},
+                    ifelse(
+                        ᶜu₃ʲ[colidx].components.data.:1 > 0,
+                        convert(BidiagonalMatrixRow{FT}, ᶜleft_bias_matrix()),
+                        convert(BidiagonalMatrixRow{FT}, ᶜright_bias_matrix()),
+                    ),
+                )
```

Comment:
Why is the convert needed here?

Reply:
Otherwise the `ifelse` isn't type-stable (it can return either a `BandMatrixRow` on the `-half` diagonal or the `+half` diagonal, depending on the velocity component).

Reply:
Oh, sorry, I thought you were asking about the inner `convert`. The outer `convert` shouldn't be needed, but I guess it reduced the allocations?

Reply:
Ah, we were trying to see if this helps with GPU. I'll remove it.